### ISSUE TITLE
Enable original package patching using swift package mirroring 

### ISSUE THREAD

**Issue by lyzkov** [johnno1962/InstantSyntax#5]
As we discussed in [_swift-syntax_](https://github.com/swiftlang/swift-syntax/issues/2421#issuecomment-2304785011), a top-level package dependency graph can be patched with:
```shell
swift package config set-mirror --original https://github.com/swiftlang/swift-syntax --mirror https://github.com/johnno1962/InstantSyntax
```

This can potentially ease _swift-syntax_ replacement with its respective binary version.
The only caveat is that the repository of the mirrored package should contain all tags from the original repository for the SPM to resolve the overridden package.

Since then, the request from the Swift community for `InstantSyntax` has been to maintain all tags from the source package repository.

Thank you in advance.

**Comment by johnno1962** [comment]
Maintaining tags for all those in the source repo isn't the only issue this repo faces. Versioning is a complex problem which Apple is much better situated to resolve given they have control of the related tooling. I'm afraid I won't be looking at this.

**Comment by lyzkov** [comment]
I don't know if all tags are necessary to recreate. Maybe major + or minor?
I’ve tried to replace the original _swift-syntax_ with my fork of _InstantSyntax_ using the package mirroring in the project that depends on __The Composable Architecture__ and it seems that a wide range of versions is available to choose from.
~~Nonetheless, the compilation failed on `switch covers known cases, but 'AccessorBlockSyntax.Accessors' may have additional unknown values`.~~

EDIT:
I've tackled the issue of nonexhaustive switch by [covering unknown cases](https://github.com/lyzkov/swift-dependencies/pull/1/commits/f74cc893663a2772da061fb8f063d1ac6eb4c2b8#) in forked versions of TCA dependencies.

However, the **main problem** is `Emit Swift module (arm64)`:
```
external macro implementation type 'PerceptionMacros.PerceptibleMacro' could not be found for macro 'Perceptible()'; '/Users/lyzkov/Library/Developer/Xcode/DerivedData/MyProject/Build/Products/Debug/PerceptionMacros' produced malformed response
```

I'm not sure because the build error message is very enigmatic. @vmanot Maybe you have some ideas?

I guess that the problem is caused by commenting out unsafe flags for weak frameworks to link.
```swift
let staticLink = ["\(clone)\(tag)/libSwiftSyntax.a"] +
    modules.map { $0.name }.flatMap({ module in
    ["-weak_framework", module] + platforms.flatMap({
        ["-F", "\(clone)\(tag)/\(module).xcframework/\($0)"] })
    })

(...)

    .target(
      name: "InstantSyntax"
//      linkerSettings: [.unsafeFlags(staticLink)]
    ),
```

I've did it get rid of package resolution error:
```
the target 'InstantSyntax' in product 'SwiftSyntaxMacros' contains unsafe build flags
```
which is (in my opinion) due to an [overly restrictive SPM policy](https://forums.swift.org/t/why-swift-package-manager-does-not-support-weak-linking-weak-framework-swiftui/31418) towards passing custom compilation flags.

@johnno1962 Have you been troubled with passing unsafe flags during _InstantSyntax_ development?

**Comment by benedelstein** [comment]
> I guess that the problem is caused by commenting out unsafe flags for weak frameworks to link.
> 
> ```swift
> let staticLink = ["\(clone)\(tag)/libSwiftSyntax.a"] +
>     modules.map { $0.name }.flatMap({ module in
>     ["-weak_framework", module] + platforms.flatMap({
>         ["-F", "\(clone)\(tag)/\(module).xcframework/\($0)"] })
>     })
> 
> (...)
> 
>     .target(
>       name: "InstantSyntax"
> //      linkerSettings: [.unsafeFlags(staticLink)]
>     ),
> ```
> 
> I've did it get rid of package resolution error:
> 
> ```
> the target 'InstantSyntax' in product 'SwiftSyntaxMacros' contains unsafe build flags
> ```
> 
> which is (in my opinion) due to an [overly restrictive SPM policy](https://forums.swift.org/t/why-swift-package-manager-does-not-support-weak-linking-weak-framework-swiftui/31418) towards passing custom compilation flags.
> 
> @johnno1962 Have you been troubled with passing unsafe flags during _InstantSyntax_ development?

@lyzkov were you able to resolve the unsafe build flag issue? It seems like the manual copying is meant to override this policy https://github.com/johnno1962/InstantSyntax/issues/3#issuecomment-1950221184 but it doesn't work for me.

**Comment by lyzkov** [comment]
@benedelstein Nope. Manual copying has nothing to do with this issue because 3rd party packages that depend on the `InstantSyntax` package are meant by SPM policies to restrict all unsafe flags from packages they depend on.

If I remember correctly, only top-level 1st party packages can have unsafe flags specified. You would have to fork all dependencies to overwrite all unsafe flags, which is an unacceptable workaround with terrible maintenance tradeoffs.

I'd suggest waiting until the SPM guys [resolve the swift-syntax prebuilt package distribution](https://github.com/swiftlang/swift-syntax/issues/2421#issuecomment-2426889975). Anyway, I hope they are not about to hack SPM's preexisting limitations through compiler flags.

**Comment by johnno1962** [comment]
I've updated he README to say this package no longer works. I've not seen it work with Xcode 16. I may archive the repo.

**Comment by johnno1962** [comment]
Seems Apple is working on a solution https://forums.swift.org/t/compilation-extremely-slow-since-macros-adoption/67921/124

**Comment by vmanot** [comment]
@johnno1962 @lyzkov we have a solution at [@PreternaturalAI](https://github.com/PreternaturalAI). See [swift-precompiled/swift-syntax](https://github.com/swift-precompiled/swift-syntax).
